accord-core/src/main/java/accord/local/Status.java

```diff
@@ -233,20 +273,63 @@ public boolean canProposeInvalidation()
             return deps.canProposeInvalidation() && executeAt.canProposeInvalidation() && outcome.canProposeInvalidation();
         }
 
+        public Known subtract(Known subtract)
+        {
+            if (!subtract.isSatisfiedBy(this))
+                return Known.Nothing;
+
+            Definition newDefinition = subtract.definition.compareTo(definition) >= 0 ? DefinitionUnknown : definition;
+            KnownExecuteAt newExecuteAt = subtract.executeAt.compareTo(executeAt) >= 0 ? ExecuteAtUnknown : executeAt;
+            KnownDeps newDeps = subtract.deps.compareTo(deps) >= 0 ? DepsUnknown : deps;
+            Outcome newOutcome = subtract.outcome.compareTo(outcome) >= 0 ? Unknown : outcome;
+            return new Known(route, newDefinition, newExecuteAt, newDeps, newOutcome);
+        }
+
+
         public String toString()
         {
             return Stream.of(definition.isKnown() ? "Definition" : null,
-                             executeAt.hasDecidedExecuteAt() ? "ExecuteAt" : null,
+                             executeAt.isDecided() ? "ExecuteAt" : null,
                              deps.hasDecidedDeps() ? "Deps" : null,
                              outcome.isOrWasApply() ? "Outcome" : null
             ).filter(Objects::nonNull).collect(Collectors.joining(",", "[", "]"));
         }
     }
 
+    public enum KnownRoute
+    {
+        /**
+         * A route may or may not be known, but it may not cover (or even intersect) this shard.
+         * The route should be relied upon only if it is a FullRoute.
+         */
+        Maybe,
```

Comment:
`Known` isn't tracked separately - and we don't reify `FullRoute` as a separate SaveStatus for each place we permit partial routes. We need to generally tighten up when/where we permit a partial route though, at which point perhaps this would be not too polluting.

Reply:
`Known` isn't tracked separately - and we don't reify `FullRoute` as a separate SaveStatus for each place we permit partial routes. We need to generally tighten up when/where we permit a partial route though, at which point perhaps this would be not too polluting.